sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -70,9 +87,15 @@ def __init__(self, id, **kwargs):
         # type: (str, Any) -> None
         self.raw_id = kwargs.get('raw_id', id)
         self.properties = CommunicationUserProperties(id=id)
+        if self.raw_id is None:
+            self.raw_id = _communication_user_raw_id(self)
 
     def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+        return self.raw_id == _communication_user_raw_id(other)
```

Comment:
`other` can be type `Any`, so we shouldn't assume it's another Identifier.
You could update this to something like:
```python
try:
    return self.raw_id == other.properties['id']
except TypeError, KeyError:
    return False
```

Reply:
Thanks! Updated